Assets/AWSIM/Scripts/Sensors/LiDAR/RglLidarPublisher.cs

```diff
@@ -78,17 +78,50 @@ private void OnEnable()
                     .AddNodePointsRos2Publish("ML_PUB", instanceIdTopic, frameID, reliabilityPolicy, durabilityPolicy, historyPolicy, historyDepth);
                 RGLNodeSequence.Connect(rglSubgraphUnity2Ros, rglSubgraphInstanceId);
             }
+
+            if (publishRadar && rglSubgraphRadar == null)
+            {
+                rglSubgraphRadar = new RGLNodeSequence()
+                    .AddNodePointsFormat("RADAR_FORMAT", FormatRadarSmartMicro.GetRGLFields())
+                    .AddNodePointsRos2Publish("RADAR_PUB", radarTopic, frameID, reliabilityPolicy, durabilityPolicy, historyPolicy, historyDepth);
+                RGLNodeSequence.Connect(rglSubgraphUnity2Ros, rglSubgraphRadar);
+            }
         }
 
         private void Start()
         {
-            if (!publishPCL24 && !publishPCL48 && !publishInstanceId)
+            if (!publishPCL24 && !publishPCL48 && !publishInstanceId && !publishRadar)
             {
                 Debug.LogWarning("All lidar message formats are disabled. Nothing to publish!");
             }
 
-            lidarSensor = GetComponent<LidarSensor>();
-            lidarSensor.ConnectToLidarFrame(rglSubgraphUnity2Ros);
+            MonoBehaviour sensor = null;
+            // Check if LiDAR is attached
+            var lidar = GetComponent<LidarSensor>();
+            if (lidar != null)
+            {
+                lidar.ConnectToLidarFrame(rglSubgraphUnity2Ros);
+                sensor = lidar;
+            }
+
+            // Check if radar is attached
+            var radar = GetComponent<RadarSensor>();
+            if (radar != null)
+            {
+                if (sensor != null)
+                {
+                    Debug.LogError($"More than one sensor is attached to the publisher. Destroying {name}.");
+                    Destroy(this);
```

Comment:
Same as for LidarSensor -> it may be worth returning after this line - setting sensor below may be not reasonable when we are destroying RglLidarPublisher. This probably refer to line 123 too.

In a specific case, setting didStart = true in line 126 below may cause bigger trouble.